src/main/java/com/example/demo/services/BookService.java

```diff
@@ -9,4 +10,6 @@
     List<Book> findAll();
 
     List<Book> findBooksContainingWord(String word);
+
+    List<Book> findBooksWithParams(BookMainInfo bookParams);
```

Comment:
Wiemy ze szukamy książek. wiec wystarczy nazwać to findBy().

Reply:
done